src/java.base/share/classes/java/util/Collections.java

```diff
@@ -1067,7 +1065,7 @@ public static <T> Collection<T> unmodifiableCollection(Collection<? extends T> c
         public String toString()                   {return c.toString();}
 
         public Iterator<E> iterator() {
-            return new Iterator<E>() {
+            return new Iterator<>() {
```

Comment:
Interesting, this whole family of changes is now possible because the original version of diamond didn't allow it to be used in the context of an anonymous inner class, because of non-denotability of types or some such. That restriction has been relaxed, but we never went back to fix these up.

I also note think that Martin objected to this sort of change in java.util.concurrent, but I'm OK with it here.

Reply:
Yes, the improved version of diamond was
JDK-8062373: Project Coin: diamond and anonymous classes
as part of JEP 213.